include/bluetooth/conn.h

```diff
@@ -413,8 +446,10 @@ struct bt_conn_cb {
 	 *
 	 *  @param conn Connection object.
 	 *  @param level New security level of the connection.
+	 *  @param reason Security error. Zero for success, non-zero otherwise.
 	 */
-	void (*security_changed)(struct bt_conn *conn, bt_security_t level);
+	void (*security_changed)(struct bt_conn *conn, bt_security_t level,
+				 u8_t err);
```

Comment:
Using u8_t for the enum seems a bit arbitrary. I think I'd rather name the enum and use it instead. That way it also becomes clearer for the user exactly what the possible values of the parameter are. You could either just use `enum bt_security_err` but considering we have `bt_security_t`perhaps it'd be more consistent to use a typedef here, i.e. `bt_security_err_t`. If you want, you could also use `__packed` with it like we do with `bt_security_t`, which essentially makes it an `u8_t` then. That said, it's not strictly needed since I don't think we use this in any structs (where memory could be saved) and operating on a word-sized integer should be more efficient (less CPU instructions).

Reply:
I agree with this, I was going to write about it but then thought I had commented enough on cosmetic issues. I don't mind typedef or enum, or even int, but u8_t definitely feels wrong.

Reply:
Yeah, makes sense, now that it is not HCI or SMP error anymore. Will switch to enum.
The typedef for bt_security_t makes sense, because it is packed. But I don't think i need packed, so will use enum instead of typedef. Seems to be more common to not typedef in zephyr.

Actually I think it is only bt_security_t and bt_addr_t, they always seems like outliers to me.